app/page/review/index.tsx

```diff
@@ -1,12 +1,46 @@
 import { useEffect, useState } from "react"
+import browser from "webextension-polyfill"
 import { useHashLocation } from "wouter/use-hash-location"
 
 import { Path } from "~app/path"
 import { useAction, usePendingTransactions } from "~app/storage"
 
 import { TransactionConfirmation } from "./transaction"
 
-function ProfileSwithcher(props: {
+export function Review() {
+  const [, navigate] = useHashLocation()
+  const pendingTxs = usePendingTransactions()
+
+  useEffect(() => {
+    async function redirect() {
+      const [tab] = await browser.tabs.query({
+        active: true,
+        lastFocusedWindow: true
+      })
+      if (tab.url?.includes("notification")) {
+        window.close()
+      } else {
+        navigate(Path.Home)
+      }
+    }
+    if (pendingTxs.length === 0) {
+      redirect()
+    }
+  }, [pendingTxs.length])
+
+  if (pendingTxs.length === 0) {
+    return
+  }
```

Comment:
For my understanding, we would redirect either to Home page or closing the widow (LN15-28). 
Do we still need this?

Reply:
After testing, this is needed.
After transaction confirmation, the `Review` component executes again, but `tx` is `undefined`, causing the `Uncaught TypeError: Cannot read properties of undefined (reading 'senderId')` error.

Or we can use the following clearer code:
```ts
  if (!tx) {
    return
  }
```

Reply:
Ah.. my bad. I didn't notice React will keep rendering while the redirection is in progress.
We do need to check again if there is a tx. 
Thanks, Irara 👍

Reply:
No problem! thanks for the thorough review. Cyan and I appreciate it! :muscle: